pkg/operator/ceph/cluster/watcher.go

```diff
@@ -53,62 +55,65 @@ func (c *clientCluster) onK8sNode(object runtime.Object) bool {
 	if !ok {
 		return false
 	}
+	// Check if the node have OSDs already present on it, reconcile if OSDs are not present
+	if alreadyPresentOSDs[node.Name] == "" {
```

Comment:
With the above suggestions:
```suggestion
	if !nodesCheckedForReconcile.Contains(node.Name) {
```